### PR TITLE
Dockerizing

### PR DESCRIPTION
**Pull request by synchrone**
I just needed a way to deploy a no-frills solution to run migrations.

ENV vars to .conf file conversion is added for docker-compose structuring

### CODE REVIEW

**Comment by mbucc** [comment]
@synchrone This is awesome in so many ways. Thanks.

**Comment by mbucc** [comment]
@synchrone I think Docker would be great for testing.

How hard is it to write a shell script that

1. spins up a docker image with a database and shmig
2. starts database if necessary
3. runs a shmig migration, capturing stdout & stderr to two different files
4. runs a shell script that verifies state

(I have very little experience with docker.)

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,30 @@
+FROM alpine
+RUN apk update
+RUN apk add bash
+RUN apk add postgresql-client
+RUN apk add mysql-client
+RUN apk add sqlite
+RUN apk add --update openssl
+
+RUN wget -O /bin/shmig https://raw.githubusercontent.com/naquad/shmig/master/shmig
```
Comment (synchrone) [inline Dockerfile]: In fact, this should have been ADD shmig /bin/shmig

**Comment by synchrone** [comment]
Regarding your usecase for testing: no, a docker container with both shmig AND the DB is probably not a good idea. This one is made from alpine (just enough linux for a container) and mysql\pgsql\sqlite clients only to run the migrations, not actually host the db.

In my usecase I have a docker-compose with pgsql as one container, and then shmig, which connects to it to run the migrations.

**Comment by mbucc** [comment]
Can you share your docker-compose script here?  I have only played a little with docker, but I understood your suggestion and it should work fine.

**Pull request merged**